File: lógica/JS/vetor/mais-velho.js
// Problema "mais_velho" 
// Fazer um programa para ler um conjunto de nomes de pessoas e suas respectivas idades. Os nomes devem ser armazenados em um vetor, e as idades em um outro vetor. Depois, mostrar na tela o nome da pessoa mais velha. 
var N, nomes = [], idades = [], maisVelho, pessoa;

N = parseInt(prompt('Quantos elementos vão ter o vetor ?'));

for(let i = 0; i < N; i++) {
    nomes[i] = prompt('Nome:');
    idades[i] = parseInt(prompt('Idade:'))
}

maisVelho = idades[0];
for(let i = 0; i < idades.length; i++) {
    if(idades[i] > maisVelho) {
        maisVelho = idades[i]
        pessoa = nomes[i];
    }
}
console.log('Pessoa mais velha: ', pessoa);

File: lógica/Python/vetor/mais_velho.py
# Problema "mais_velho" 
# Fazer um programa para ler um conjunto de nomes de pessoas e suas respectivas idades. 
# Os nomes devem ser armazenados em um vetor, e as idades em um outro vetor.
#  Depois, mostrar na tela o nome da pessoa mais velha.
maisVelho: int; nome: str
n = int(input("Quantas pessoas voce vai digitar ? "))
vetNome = [0 for x in range(n)]
vetIdade = [0 for x in range(n)]

for i in range(n):
    print(f"Dados da {i+1}º pessoa: ")
    vetNome[i] = input("Nome: ")
    vetIdade[i] = int(input("Idade: "))

maisVelho = vetIdade[0]
nome = vetNome[0]
for i in range(n):
    if vetIdade[i] > maisVelho:
        maisVelho = vetIdade[i]
        nome = vetNome[i]

print(f"Pessoa mais velha = {nome}")